src/main/java/com/questhelper/quests/treegnomevillage/TreeGnomeVillage.java

```diff
@@ -0,0 +1,207 @@
+		steps.put(4, talkToTrackersStep());
+		steps.put(5, retrieveOrbStep());
+		steps.put(6, talkToKingBolrenFirstOrb);
+		steps.put(7, defeatWarlordStep());
+		steps.put(8, returnOrbs);
+		return steps;
+	}
+
+	private QuestStep talkToTrackersStep()
+	{
+		ConditionalStep talkToTrackers = new ConditionalStep(this, firstTracker);
+		talkToTrackers.addStep(talkToSecondTracker, secondTracker);
+		talkToTrackers.addStep(talkToThirdTracker, thirdTracker);
+		talkToTrackers.addStep(shouldFireBallista, fireBallista);
+		return talkToTrackers;
+	}
+
+	private QuestStep retrieveOrbStep()
+	{
+		retrieveOrb = new ConditionalStep(this, climbTheLadder, "Retrieve the first orb from chest.");
+		ObjectStep getOrbFromChest = new ObjectStep(this, 2183, new WorldPoint(2506, 3259, 1), "Retrieve the first orb from chest.");
+		retrieveOrb.addStep(isUpstairsTower, getOrbFromChest);
+		retrieveOrb.addSubSteps(getOrbFromChest, climbTheLadder);
+		return retrieveOrb;
+	}
+
+	private QuestStep defeatWarlordStep()
+	{
+		NpcCondition fightingWarlord = new NpcCondition(NpcID.KHAZARD_WARLORD_7622);
+
+		fightTheWarlord = new NpcStep(this, NpcID.KHAZARD_WARLORD_7622, new WorldPoint(2456, 3301, 0), "Defeat the warlord and retrieve orbs");
+		talkToTheWarlord = new NpcStep(this, NpcID.KHAZARD_WARLORD_7621, new WorldPoint(2456, 3301, 0), "Retrieve the orbs from the warlord.");
+
+		ConditionalStep defeatTheWarlord = new ConditionalStep(this, talkToTheWarlord,
+			new ItemRequirement("Food", -1),
+			new ItemRequirement("Armor & Weapons", -1));
+		defeatTheWarlord.addStep(fightingWarlord, fightTheWarlord);
+
+		return defeatTheWarlord;
+	}
+
+	private void setupItemRequirements()
+	{
+		logRequirement = new ItemRequirement("Logs", ItemID.LOGS, 6);
+	}
+
+	private void setupZones()
+	{
+		upstairsTower = new Zone(new WorldPoint(2500, 3251, 1), new WorldPoint(2506, 3259, 1));
+	}
+
+	public void setupConditions()
+	{
+		notCompleteFirstTracker = new VarbitCondition(TRACKER_1_VARBITID, 0);
+		notCompleteSecondTracker = new VarbitCondition(TRACKER_2_VARBITID, 0);
+		notCompleteThirdTracker = new VarbitCondition(TRACKER_3_VARBITID, 0);
+
+		completeFirstTracker = new VarbitCondition(TRACKER_1_VARBITID, 1);
+		completeSecondTracker = new VarbitCondition(TRACKER_2_VARBITID, 1);
+		completeThirdTracker = new VarbitCondition(TRACKER_3_VARBITID, 1);
+
+		isUpstairsTower = new ZoneCondition(upstairsTower);
+
+		talkToFirstTracker = new Conditions(notCompleteFirstTracker);
+		talkToSecondTracker = new Conditions(LogicType.AND, completeFirstTracker, notCompleteSecondTracker);
+		talkToThirdTracker = new Conditions(LogicType.AND, completeFirstTracker, notCompleteThirdTracker);
+		shouldFireBallista = new Conditions(LogicType.AND, completeFirstTracker, completeSecondTracker, completeThirdTracker);
+	}
+
+	private void setupSteps()
+	{
+		talkToKingBolren = new NpcStep(this, NpcID.KING_BOLREN, new WorldPoint(2541, 3170, 0), "Speak to King Bolren in the centre of the Tree Gnome Maze.");
```

Comment:
This step would really benefit with some world lines guiding players through the maze. This can be done with something like the following:

```
ArrayList<WorldPoint> pathThroughMaze = new ArrayList<>(Arrays.asList(
			new WorldPoint(3222, 9618, 0),
			new WorldPoint(3224, 9618, 0));
talkToKingBolren.setLinePoints(pathThroughMaze);
```